http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -992,18 +1006,26 @@ private void executeRoute() {
             AtomicReference<io.micronaut.http.HttpRequest<?>> requestReference = new AtomicReference<>(request);
             boolean isFuture = CompletionStage.class.isAssignableFrom(javaReturnType);
             boolean isReactiveReturnType = Publishers.isConvertibleToPublisher(javaReturnType) || isFuture;
+            boolean isKotlinSuspendingFunction =
+                    finalRoute instanceof MethodBasedRouteMatch &&
+                    isKotlinSuspendingFunction(((MethodBasedRouteMatch) finalRoute).getTargetMethod());
+            boolean isKotlinFunctionReturnTypeUnit = isKotlinSuspendingFunction &&
+                    isKotlinFunctionReturnTypeUnit(((MethodBasedRouteMatch) finalRoute).getTargetMethod());
```

Comment:
Yeah these usages of `getTargetMethod()` are problematic as per my previous comment.

I also think that in general this is starting get messy. Although that is not the fault of the PR but more Micronaut's lack of an area to plug into. We can probably roll with it for now, but need to revisit this area to improve how this is handled.

Reply:
Got rid of `getTargetMethod()` (see my comment above).

I agree that the code started to be difficult to manage and extend and some refactoring and adding some ability to make it more pluggable would be nice.

The other thing which worries me is the code in `DefaultRequestBinderRegistry` which I stumbled upon while creating this PR. What I mean exactly is the `Map<Integer, RequestArgumentBinder> byType` solution since in the case of hash collision (which is the key to this map) we can get into trouble...

Reply:
Yeah I agree, this can probably just use the class as the key

Reply:
Though I would do it in a separate PR.